diff --git a/src/server/game/Entities/Player/Player.cpp b/src/server/game/Entities/Player/Player.cpp
index f3d45b0..c29fb3b 100644
--- a/src/server/game/Entities/Player/Player.cpp
+++ b/src/server/game/Entities/Player/Player.cpp
@@ -4652,6 +4652,14 @@ void Player::BuildPlayerRepop()
     SetMovement(MOVE_WATER_WALK);
     if (!GetSession()->isLogingOut())
         SetMovement(MOVE_UNROOT);
+	
+    // Ghost Flying Mount
+#define GhostMount_DisplayID	     31154 // Icebound Frostbrood Vanquisher
+#define Flying_Speed			4.10f // 310% Flying Speed. You can edit to 3.80f = 280% flying speed. Or 2.50f = 150%
+#define Ground_Speed			2.0f  // 100% Ground Speed
+
+	SetUInt32Value(UNIT_FIELD_MOUNTDISPLAYID, GhostMount_DisplayID);
+	SetSpeed(MOVE_RUN, Ground_Speed, true); // Ground mount Speed
+	SetCanFly(true);
+	SetSpeed(MOVE_FLIGHT, Flying_Speed, true); // Flying mount Speed
 
     // BG - remove insignia related
     RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_SKINNABLE);
@@ -4696,6 +4704,11 @@ void Player::ResurrectPlayer(float restore_percent, bool applySickness)
 
     SetMovement(MOVE_LAND_WALK);
     SetMovement(MOVE_UNROOT);
+	
+	//dismount upon resurrection
+	SetUInt32Value(UNIT_FIELD_MOUNTDISPLAYID, 0);
+	//disable flight
+	SetCanFly(false, true);
 
     m_deathTimer = 0;
 
